sdk/keyvault/keyvault-admin/test/utils/authentication.ts

```diff
@@ -56,18 +58,20 @@ export async function authenticate(that: any): Promise<any> {
     queryParametersToSkip: []
   };
   const recorder = record(that, recorderEnvSetup);
+
   const credential = new ClientSecretCredential(
-    env.AZURE_TENANT_ID,
-    env.AZURE_CLIENT_ID,
-    env.AZURE_CLIENT_SECRET
+    getEnvironmentVariable("AZURE_TENANT_ID"),
+    getEnvironmentVariable("AZURE_CLIENT_ID"),
+    getEnvironmentVariable("AZURE_CLIENT_SECRET")
   );
 
-  const keyVaultHsmUrl = env.AZURE_MANAGEDHSM_URL;
-  if (!keyVaultHsmUrl) {
-    throw new Error("Missing AZURE_MANAGEDHSM_URL environment variable.");
-  }
+  const keyVaultHsmUrl = getEnvironmentVariable("AZURE_MANAGEDHSM_URI");
 
-  const accessControlClient = new KeyVaultAccessControlClient(keyVaultHsmUrl, credential);
+  // Passing a separate httpClient for every instance as a workaround
+  // for a caching issue when creating role assignments
+  const accessControlClient = new KeyVaultAccessControlClient(keyVaultHsmUrl, credential, {
+    httpClient: new DefaultHttpClient()
+  });
```

Comment:
Though I love that we can do this, please get in touch with @xirzec regarding why this `httpClient` is needed 🙏. This could be seen as a bug - if that's the case, we should treat it as a core bug rather than looking for ways to bypass it.

Reply:
Yes thanks for the reminder 😄  I reached out to him and will show what we ran into!